### PR TITLE
skills: add special skill type for testing

### PR DESCRIPTION
**Pull request by zyga**
Many tests require a valid type to work with. The TestType is exactly
that, allowing us to create a range of valid and invalid capabilities,
depending on the needs of a particular test.

### CODE REVIEW

File: skills/testtype.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+// Name returns the name of the test type.
+func (t *TestType) Name() string {
+	return t.TypeName
+}
+
+// Sanitize checks and possibly modifies a skill.
+func (t *TestType) Sanitize(skill *Skill) error {
+	if t.Name() != skill.Type {
+		return fmt.Errorf("skill is not of type %q", t)
```
Comment (niemeyer) [inline skills/testtype.go]: This should be a panic. A Type should never-ever receive an unknown type to check.

File: skills/testtype.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	if t.SanitizeCallback != nil {
+		return t.SanitizeCallback(skill)
+	}
+	return nil
+}
+
+// SecuritySnippet returns the configuration snippet "required" to use a test skill.
+// Consumers don't gain any extra permissions.
+func (t *TestType) SecuritySnippet(skill *Skill, securitySystem SecuritySystem) ([]byte, error) {
+	return nil, nil
```
Comment (niemeyer) [inline skills/testtype.go]: Might be worth having a callback here too at some point, same style as Sanitize.

**Comment by niemeyer** [comment]
LGTM with a few trivials for you consideration.

File: skills/testtype.go (hunk 1 above)
Comment (zyga) [inline skills/testtype.go]: Makes sense, changed.

File: skills/testtype.go (hunk 2 above)
Comment (zyga) [inline skills/testtype.go]: Yes but given the tricky nature of security (our online discussion of producer and consumer side security snippets) I'd like to revisit this separately.

**Comment by mvo5** [comment]
Looks good

**Pull request merged**